pkg/imgutil/imgutil.go

```diff
@@ -51,8 +54,15 @@ type EnsuredImage struct {
 	Remote      bool // true for stargz or overlaybd
 }
 
-// PullMode is either one of "always", "missing", "never"
-type PullMode = string
+type Config struct {
+	Stdout           io.Writer
+	Stderr           io.Writer
+	Snapshotter      string
+	OcispecPlatforms []ocispec.Platform
+	Unpack           *bool
+	Quiet            bool
+	RFlags           RemoteSnapshotterFlags
+}
```

Comment:
Can we consolidate this with `ImagePullOptions`?

https://github.com/containerd/nerdctl/blob/78b66fdcde0eeafb95fdf9915dc4ccbaef51021a/pkg/api/types/image_types.go#L185

Reply:
I recall having tried this in a [previous PR](https://github.com/containerd/nerdctl/pull/2638#discussion_r1393504324) where I implemented the `--soci-index-digest` flag and ended up creating a circular dependency. Per my comment linked it was due to the `netutil` package creating a circular dependency.

I think the idea is logically sound, and I don't believe this circular dependency is necessary. I've created a [new PR](https://github.com/containerd/nerdctl/pull/2892) to hopefully remove any chance circular dependencies. If it gets merged I can rebase this PR and use the options struct instead of this new type.